core/indexer/common.go

```diff
@@ -448,15 +412,62 @@ func prepareTxUpdate(tx *Transaction) ([]byte, []byte) {
 
 	if tx.GasUsed == tx.GasLimit {
 		// do not update gasUsed because it is the same with gasUsed when transaction was saved first time in database
-		serializedData = []byte(fmt.Sprintf(`{ "doc" : { "log" : %s, "scResults" : %s, "status": "%s", "timestamp": %s } }`,
-			string(marshalizedLog), string(scResults), tx.Status, string(marshalizedTimestamp)))
+		serializedData =
```

Comment:
a lot of duplicated code here.

Reply:
I know but I couldn't find another approach to include an additional field by using the same base string (a substring has to be added inside `script` part and inside `params` as well)

Reply:
I think we should have a separate task here. This json encoding for transaction serialisation became complicated enough that it deserves a component. Could be a simple structure with some helper methods that can receive a list of fields to update, a list of fields for insert, etc. It can have json tags on it's fields and then we can just marshal it